main.tf

```diff
@@ -3,13 +3,103 @@ terraform {
     organization = "ffddorf"
 
     workspaces {
-      prefix = "app-k8s-template-"
+      prefix = "app-k8s-template-" # Change to app-your-thing-
     }
   }
 }
 
-resource "kubernetes_namespace" "demo" {
+# If applied as is, this template would put up an empty Nginx site at https://k8s-template.dev.dorf.world/
+
+locals {
+  app_name = "k8s-template" # Change to your-thing
+  host = "k8s-template.dev.dorf.world" # Change to your-thing.dev.dorf.world
+}
+
+resource "kubernetes_namespace" "app" {
+  metadata {
+    name = "app-k8s-template"
+  }
+}
+
+resource "kubernetes_deployment" "app" {
+  metadata {
+    name = local.app_name
+    namespace = kubernetes_namespace.app.metadata[0].name
+    labels = {
+      k8s_app = local.app_name
+    }
+  }
+  spec {
+    selector {
+      match_labels = {
+        k8s_app = local.app_name
+      }
+    }
+    template {
+      metadata {
+        labels = {
+          k8s_app = local.app_name
+        }
+      }
+      spec {
+        container {
+          name = local.app_name
+          image = "nginx"
+        }
+      }
+    }
+  }
+}
+
+resource "kubernetes_service" "app" {
   metadata {
-    name = "demo"
+    name = local.app_name
+    namespace = kubernetes_namespace.app.metadata[0].name
+    labels = {
+      service = local.app_name
+    }
+  }
+  spec {
+    port {
+      name = "http"
+      port = 80
+      target_port = 80
+    }
+    selector = {
+      k8s_app = local.app_name
+    }
+    type = "NodePort"
```

Comment:
doesn't need to be exposed on a port, let's not have this in the example

Reply:
You mean just get rid of that one line?

Reply:
```suggestion
    type = "ClusterIP"
```